### PR TITLE
semaphoreci integration

### PR DESCRIPTION
**Pull request by soerenmartius**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -4,11 +4,16 @@ MAINTAINER "The Mineiros.io Team <hello@mineiros.io>"
...
-RUN apk add --update bash curl git python3 terraform
+RUN apk add --update bash curl git openssl python3 terraform
 
 RUN pip3 install pre-commit
 
-RUN curl -L -o /tmp/tflint.zip https://github.com/terraform-linters/tflint/releases/download/${TFLINT_VERSION}/tflint_linux_amd64.zip && unzip /tmp/tflint.zip -d /usr/local/bin
+ADD https://github.com/terraform-linters/tflint/releases/download/${TFLINT_VERSION}/tflint_linux_amd64.zip .
+ADD https://github.com/terraform-linters/tflint/releases/download/${TFLINT_VERSION}/checksums.txt .
+RUN sed -i '/.*tflint_linux_amd64.zip/!d' checksums.txt
+RUN sha256sum -cs checksums.txt
```
Comment (mariux) [inline Dockerfile]: that's a weird check as it's the same source ;) ... if i can manipulate the zip i can also update the checksum.. ;)

File: .semaphore/semaphore.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      prologue:
+        commands:
+          - export BUILD_VERSION=${SEMAPHORE_GIT_SHA:0:7}
+          - checkout
+      jobs:
+        - name: "Build Docker Image"
+          commands:
+            - make docker-build
+            - make docker-save
+            - cache store docker-image-$BUILD_VERSION terraform-github-organization-$BUILD_VERSION.tar
```
Comment (mariux) [inline .semaphore/semaphore.yml]: is there a way to get this from a public docker registry?

Comment (soerenmartius) [reply .semaphore/semaphore.yml]: I haven't researched it yet but I guess so. That would also be one of the next logical steps to me.

File: Dockerfile (hunk 1 above)
Comment (soerenmartius) [reply Dockerfile]: Hmm, how would you do that normally? In our case, both, the released zip file and the txt containing the checksum are hosted on Github. I haven't seen cases, where people would upload one or another to a different source ?

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+docker-run-pre-commit-hooks: docker-build
+	docker run --rm ${REPOSITORY_NAME}:${BUILD_VERSION} pre-commit run --all-files
+
+# Run pre-commit hooks using a cached image
+docker-run-pre-commit-hooks-from-cache: docker-load
+	docker run --rm  ${REPOSITORY_NAME}:${BUILD_VERSION} pre-commit run --all-files
+
+# run tests
+docker-run-tests: docker-build
+	docker run --rm -e GITHUB_TOKEN=${GITHUB_TOKEN} -e GITHUB_ORGANIZATION=${GITHUB_ORGANIZATION} ${REPOSITORY_NAME}:${BUILD_VERSION} go test -v -timeout 30m test/github_organization_test.go
```
Comment (mariux) [inline Makefile]:
```suggestion
	docker run --rm -e GITHUB_TOKEN -e GITHUB_ORGANIZATION ${REPOSITORY_NAME}:${BUILD_VERSION} go test -v -timeout 30m test/github_organization_test.go
```

shoudl do the job

Comment (soerenmartius) [reply Makefile]: ah cool, didn't know that!

**Pull request merged**